crates/epaint/src/text/text_layout.rs

```diff
@@ -129,7 +129,7 @@ fn layout_section(
             });
 
             paragraph.cursor_x += glyph_info.advance_width;
-            paragraph.cursor_x = font.round_to_pixel(paragraph.cursor_x);
+            paragraph.cursor_x = paragraph.cursor_x.round();
```

Comment:
This rounds each glyph to point coordinates. The previous code rounded it to pixel coordinates.

The reason we want to round it to pixel coordinates is that it creates more crisp text. Each glyph is rasterized to a little bitmap (part of the font-atlas), and rounding to pixel coordinates ensures that we don't get any billinear filtering-blur when reading them.

In other words: we have a trade-off to make. Do we want crisp text, or well-kerned text? We can't really have both with the current font rendering technique.

Rounding to points coordiantes makes no sense at all, and will just create _worse_ kerning whenever pixels_per_point is non-integral.

For instance, here is the demo app at 0.95 scale, before:

![font-0 95-before](https://user-images.githubusercontent.com/1148717/213987630-b4c526ab-458b-4cc4-821a-1b246792b300.png)

And this PR:
![font-0 95-after](https://user-images.githubusercontent.com/1148717/213987665-c7036d99-7ca2-4604-ad66-a206d147de24.png)

You can notice that a lot of the text is blurrier (especially visible in the monospace code) and that the kerning is worse in many places.

Reply:
My understanding is that this specific line only changes the kerning on glyphs. Rasterization is rounded to the pixel grid in `tessellate_glyphs()`, which is not changed by this PR. It isn't clear to me that rounding here is useful at all, but I kept it (in point space) because it does improve the kerning at some scales.

But you can just as well `floor`, `ceil` or even remove the rounding entirely on this line and it will not affect glyph clarity or the pixel grid in the tessellator.